x-pack/test/functional/services/ml/stack_management_jobs.ts

```diff
@@ -99,45 +97,38 @@ export function MachineLearningStackManagementJobsProvider(
     },
 
     async assertADJobRowSpaces(adJobId: string, expectedSpaces: string[]) {
-      await mlADJobTable.refreshJobList('stackMgmtJobList');
-      const rows = await mlADJobTable.parseJobTable('stackMgmtJobList');
-      const jobRow = rows.filter((row) => row.id === adJobId)[0];
-      expect(jobRow).to.have.property('spaces');
-      expect(jobRow.spaces!.sort()).to.eql(
+      await this.refreshList();
+      const spaces = await this.getSpacesFromTable(adJobId);
+      expect(spaces!.sort()).to.eql(
         expectedSpaces.sort(),
         `Expected spaces for AD job '${adJobId}' to be '${JSON.stringify(
           expectedSpaces
-        )}' (got '${JSON.stringify(jobRow.spaces)}')`
+        )}' (got '${JSON.stringify(spaces)}')`
       );
     },
 
     async assertDFAJobRowSpaces(dfaJobId: string, expectedSpaces: string[]) {
-      await mlDFAJobTable.refreshAnalyticsTable('stackMgmtJobList');
-      const rows = await mlDFAJobTable.parseAnalyticsTable('stackMgmtJobList');
-      const jobRow = rows.filter((row) => row.id === dfaJobId)[0];
-      expect(jobRow).to.have.property('spaces');
-      expect(jobRow.spaces!.sort()).to.eql(
+      await this.refreshList();
+      const spaces = await this.getSpacesFromTable(dfaJobId);
+      expect(spaces!.sort()).to.eql(
         expectedSpaces.sort(),
         `Expected spaces for DFA job '${dfaJobId}' to be '${JSON.stringify(
           expectedSpaces
-        )}' (got '${JSON.stringify(jobRow.spaces)}')`
+        )}' (got '${JSON.stringify(spaces)}')`
       );
     },
 
-    async openADJobSpacesFlyout(adJobId: string) {
+    async refreshList() {
```

Comment:
The selected page isn't being retained after editing the spaces for a job e.g. Edit job on page 2, save changes, and table refreshes to show page 1. On the current table the page is retained.

Reply:
Added table state in https://github.com/elastic/kibana/pull/134893/commits/ac13e3713702c475db1aee0b9601cb2ccc3395e0